### PR TITLE
Region migration related work

### PR DESCRIPTION
**Pull request by liyuheng55555**
- IConsensus interface idempotence
- Clear uncompleted region file when DataNode restart
- IT framework enhancement
- More test cases

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 1]
```diff
@@ -450,6 +457,34 @@ private void sendRegisterRequestToConfigNode() throws StartupException, IOExcept
...
+    return consensusGroupIds;
+  }
+
+  private void clearInvalidConsensusGroup(List<ConsensusGroupId> configNodeConsensusGroupIds) {
+    List<ConsensusGroupId> dataNodeConsensusGroupIds = getConsensusGroupId();
+    String dataRegionConsensusDir = config.getDataRegionConsensusDir();
+    for (ConsensusGroupId consensusGroupId : dataNodeConsensusGroupIds) {
+      if (!configNodeConsensusGroupIds.contains(consensusGroupId)) {
+        String path = IoTConsensus.buildPeerDir(new File(dataRegionConsensusDir), consensusGroupId);
+        FileUtils.deleteDirectory(new File(path));
```
Comment (liyuheng55555) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:483]: Maybe rename dir, or move dir to somewhere else ?

File: iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties [hunk 2]
```diff
@@ -21,10 +21,10 @@ dn_rpc_address=127.0.0.1
...
 
 dn_rpc_port=6668
-dn_internal_port=10731
-dn_mpp_data_exchange_port=10741
-dn_schema_region_consensus_port=10751
-dn_data_region_consensus_port=10761
+dn_internal_port=10831
+dn_mpp_data_exchange_port=10841
+dn_schema_region_consensus_port=10851
+dn_data_region_consensus_port=10861
```
Comment (liyuheng55555) [inline iotdb-core/datanode/src/test/resources/datanode2conf/iotdb-datanode.properties:27]: revert ?

**Review by liyuheng55555 — COMMENTED** [review]
PTAL~  @BUAAserein

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 3]
```diff
@@ -450,6 +458,100 @@ private void sendRegisterRequestToConfigNode() throws StartupException, IOExcept
     }
   }
 
+  private List<ConsensusGroupId> getConsensusGroupId() {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:461]: TODO

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java [hunk 4]
```diff
@@ -400,6 +402,12 @@ public TDataNodeRestartResp restartDataNode(TDataNodeRestartReq req) throws TExc
         () -> client.restartDataNode(req), resp -> !updateConfigNodeLeader(resp.status));
   }
 
+  @Override
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java:405]: remove

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java [hunk 5]
```diff
@@ -255,6 +257,16 @@ public TDataNodeRestartResp restartDataNode(TDataNodeRestartReq req) {
     return resp;
   }
 
+  @Override
+  public TDataNodeRemoveRegionResp removeRegion(TDataNodeRemoveRegionReq req) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java:261]: remove

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/IManager.java [hunk 6]
```diff
@@ -254,6 +256,14 @@ public interface IManager {
    */
   TDataNodeRestartResp restartDataNode(TDataNodeRestartReq req);
 
+  /**
+   * Remove Invalid Region.
+   *
+   * @param req TDataNodeRemoveRegionReq
+   * @return {@link TSStatusCode#SUCCESS_STATUS} if submit RemoveRegionPeerProcedure success.
+   */
+  TDataNodeRemoveRegionResp removeRegion(TDataNodeRemoveRegionReq req);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/IManager.java:265]: remove

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Add todo for all unfinished list

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**